diff --git a/src/base/codec_types.h b/src/base/codec_types.h
index e634015..ec565e1 100644
--- a/src/base/codec_types.h
+++ b/src/base/codec_types.h
@@ -217,7 +217,7 @@ std::string FourccToString(uint32_t fourcc);
 VideoCodec VideoCodecProfileToVideoCodec(VideoCodecProfile profile);

 typedef std::vector<SupportedProfile> SupportedProfiles;
-typedef std::chrono::_V2::system_clock::time_point ChronoTime;
+typedef std::chrono::system_clock::time_point ChronoTime;

 }  //  namespace mcil
