From 8e7fa6409bc835c0e3e5f54b4837f29c6bc9df79 Mon Sep 17 00:00:00 2001
From: Jaekyu Lee <jaekyu.lee@lge.com>
Date: Mon, 19 Oct 2020 20:04:15 +0900
Subject: [PATCH] Fix return-type error

---
 mkdosfs/mkdosfs.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mkdosfs/mkdosfs.c b/mkdosfs/mkdosfs.c
index caeb8f5..b352a0c 100644
--- a/mkdosfs/mkdosfs.c
+++ b/mkdosfs/mkdosfs.c
@@ -1551,7 +1551,7 @@ static int add_file(char *filename, struct dir_entry *dir, unsigned char attr)
   if (dir->root) {
     if (dir->count == dir->entries) {
       printf("Error - too many directory entries\n");
-      return;
+      return -1;
     }
   }
   else {
@@ -1570,7 +1570,7 @@ static int add_file(char *filename, struct dir_entry *dir, unsigned char attr)
   }
 
   infile = open(filename, O_RDONLY, 0);
-  if (!infile) return;
+  if (!infile) return -1;
   
   if (fstat(infile, &stat))
     goto exit_add;
-- 
2.17.1
