From 9888f1084506eb02f4c30078c60080f1967a4dc0 Mon Sep 17 00:00:00 2001
From: "sungmok.shin" <sungmok.shin@lge.com>
Date: Mon, 20 Apr 2020 19:24:45 +0900
Subject: [PATCH] Set default pairing capability as "NoInputNoOutput"

---
Upstream-Status: Pending

 src/agent.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/agent.c b/src/agent.c
index 7d66cf50d..849a527c8 100644
--- a/src/agent.c
+++ b/src/agent.c
@@ -979,7 +979,7 @@ static DBusMessage *register_agent(DBusConnection *conn,
 						DBUS_TYPE_INVALID) == FALSE)
 		return btd_error_invalid_args(msg);
 
-	cap = parse_io_capability(capability);
+	cap = parse_io_capability("NoInputNoOutput");
 	if (cap == IO_CAPABILITY_INVALID)
 		return btd_error_invalid_args(msg);
 
