diff -ruN mtpfs-1.1/mtpfs.c mtpfs-1.1-new/mtpfs.c
--- mtpfs-1.1/mtpfs.c	2012-02-27 09:59:39.000000000 +0900
+++ mtpfs-1.1-new/mtpfs.c	2013-11-05 15:39:59.798640666 +0900
@@ -649,7 +649,7 @@
 
     // If in root directory
     if (strcmp(path,"/") == 0) {
-        filler (buf, "Playlists", NULL, 0);
+        //filler (buf, "Playlists", NULL, 0);
 		if (lostfiles != NULL) {
 			filler (buf, "lost+found", NULL, 0);
 		}
@@ -725,12 +725,12 @@
     if (folder_id==-2) {
         DBG("Root of storage area");
         folder=storageArea[storageid].folders;
+    	if (folder == NULL) return_unlock(0);
     } else {
         folder = LIBMTP_Find_Folder (storageArea[storageid].folders, folder_id);
         if (folder == NULL) return_unlock(0);
         folder = folder->child;
     }
-    if (folder == NULL) return_unlock(0);
 
     while (folder != NULL) {
         if ((folder->parent_id == folder_id) ||
