From ad7c3f9b795959f76376dae0b36342e7c1c4585d Mon Sep 17 00:00:00 2001
From: "piyush10.asalmol" <piyush10.asalmol@lge.com>
Date: Tue, 11 Oct 2022 23:43:53 +0530
Subject: [PATCH] Fix-online-state-change

---
 src/service.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/service.c b/src/service.c
index ef67d31..9fda6f6 100644
--- a/src/service.c
+++ b/src/service.c
@@ -6617,6 +6617,11 @@ int __connman_service_ipconfig_indicate_state(struct connman_service *service,
 	if (!is_connected(old_state) && is_connected(new_state))
 		nameserver_add_all(service, type);
 
+	if(new_state==CONNMAN_SERVICE_STATE_READY) {
+		service->state_ipv4 = CONNMAN_SERVICE_STATE_ONLINE;
+		state_changed(service);
+	}
+
 	__connman_timeserver_sync(service);
 
 	return service_indicate_state(service);
-- 
2.17.1

