diff --git a/daemon/Makefile b/daemon/Makefile
index 9cd04ce..57f10f7 100644
--- a/daemon/Makefile
+++ b/daemon/Makefile
@@ -24,8 +24,6 @@ ifeq ($(shell expr `$(CXX) -dumpversion | cut -f1 -d.` \>= 5),1)
 	CXXFLAGS += -fno-sized-deallocation
 endif
 
-# -s strips the binary of debug info
-LDFLAGS     += -s
 LDLIBS      += -lrt -lm -pthread
 TARGET      := $(OBJ_DIR)gatord
 ESCAPE_EXE  := $(OBJ_DIR)escape/escape
