From 35e78acc1346856e18e36d8414c21c48d136d164 Mon Sep 17 00:00:00 2001
From: "eugene.todoruk" <eugene.todoruk@lge.com>
Date: Wed, 29 May 2019 12:32:32 +0300
Subject: [PATCH] meson: add smack-default-process-label option

---
 meson.build       | 5 +++++
 meson_options.txt | 2 ++
 2 files changed, 7 insertions(+)

diff --git a/meson.build b/meson.build
index e3f2a2aee7..772524a3d0 100644
--- a/meson.build
+++ b/meson.build
@@ -929,6 +929,11 @@ if smack_run_label != ''
         conf.set_quoted('SMACK_RUN_LABEL', smack_run_label)
 endif
 
+smack_default_process_label = get_option('smack-default-process-label')
+if smack_default_process_label != ''
+        conf.set_quoted('SMACK_DEFAULT_PROCESS_LABEL', smack_default_process_label)
+endif
+
 want_polkit = get_option('polkit')
 install_polkit = false
 install_polkit_pkla = false
diff --git a/meson_options.txt b/meson_options.txt
index 044bb792b5..48d04ca0f3 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -234,6 +234,8 @@ option('smack', type : 'boolean',
        description : 'SMACK support')
 option('smack-run-label', type : 'string',
        description : 'run systemd --system itself with a specific SMACK label')
+option('smack-default-process-label', type : 'string',
+       description : 'run systemd services with a specific SMACK label')
 option('polkit', type : 'combo', choices : ['auto', 'true', 'false'],
        description : 'polkit support')
 option('ima', type : 'boolean',
