--- sdparm-1.08/src/sdparm.c	2013-06-06 22:59:59.000000000 +0900
+++ sdparm-1.08-new/src/sdparm.c	2013-11-28 15:59:38.678254090 +0900
@@ -1744,7 +1744,7 @@
             sg_cmds_close_device(sg_fd);
             sg_fd = sg_sg_fd;
             res = sg_simple_inquiry(sg_fd, &sir, 0, verb);
-            if (sg_sg_fd < 0)
+            if (res < 0)
                 goto err_out;
         }
 #endif  /* SG_LIB_LINUX */
@@ -1899,6 +1899,7 @@
             break;
         case 'C':
             cmd_str = optarg;
+	     break;
         case 'd':
             ++opts.dummy;
             break;
@@ -2453,12 +2454,14 @@
         if (err < 0) {
             if (verbose)
                 perror("SCSI_IOCTL_GET_BUS_NUMBER failed");
+	     close(fd);
             return -2;
         }
         err = ioctl(fd, SCSI_IOCTL_GET_IDLUN, &mm_idlun);
         if (err < 0) {
             if (verbose)
                 perror("SCSI_IOCTL_GET_IDLUN failed");
+	     close(fd);
             return -2;
         }
         if ((bus == bbus) &&
@@ -2476,6 +2479,7 @@
     if (0 == num_sgdevs) {
         fprintf(stderr, "No /dev/sg* devices found; is the sg driver "
                 "loaded?\n");
+	 close(fd);
         return -2;
     }
     if (fd >= 0) {
