diff --git a/src/pstree.c b/src/pstree.c
index 071e6c4..4f13b31 100644
--- a/src/pstree.c
+++ b/src/pstree.c
@@ -816,10 +816,10 @@ static char* get_threadname(const pid_t pid, const int tid, const char *comm)
 		return threadname;
 	    }
 	}
+	fclose(file);
     }
     /* Fall back to old method */
     sprintf(threadname, "{%.*s}", COMM_LEN, comm);
-    fclose(file);
     return threadname;
 }
 
