diff -ruN a/configure.ac b/configure.ac
--- a/configure.ac	2019-11-13 14:32:21.886869415 +0530
+++ b/configure.ac	2019-11-13 14:32:43.975365442 +0530
@@ -254,7 +254,7 @@
 fi
 
 if test "x$ac_cv_omx_target" = "xrpi"; then
-    PKG_CHECK_MODULES([BRCMEGL], [brcmegl])
+    PKG_CHECK_MODULES([EGL], [egl])
 fi
 
 AC_ARG_WITH([omx-header-path],
diff -ruN a/examples/egl/Makefile.am b/examples/egl/Makefile.am
--- a/examples/egl/Makefile.am	2019-11-13 14:32:22.022872469 +0530
+++ b/examples/egl/Makefile.am	2019-11-13 14:32:44.011366251 +0530
@@ -18,7 +18,7 @@
 	$(GST_BASE_LIBS) \
 	$(GST_LIBS) \
 	$(GST_GL_LIBS) \
-	$(BRCMEGL_LIBS) \
+	$(EGL_LIBS) \
 	-lm
 
 if HAVE_GLES2
@@ -38,4 +38,4 @@
 	$(GST_BASE_CFLAGS) \
 	$(GST_CFLAGS) \
 	$(GST_GL_CFLAGS) \
-	$(BRCMEGL_CFLAGS)
+	$(EGL_CFLAGS)
