From 2029c1bf384ef97258ac8248e444257d597f6400 Mon Sep 17 00:00:00 2001
From: Arun Raghavan <arun@asymptotic.io>
Date: Tue, 19 Oct 2021 16:20:35 -0400
Subject: [PATCH] Add missing absl library for bad_optional_access

---
Upstream-Status: Backport [1.1 https://gitlab.freedesktop.org/pulseaudio/webrtc-audio-processing/-/commit/0cc2ebeda2faf60f3367bd5e88e1247455dfcfee]

 meson.build | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 35decd3..eadf282 100644
--- a/meson.build
+++ b/meson.build
@@ -48,6 +48,7 @@ absl_base_dep = dependency('absl_base', required : false)
 absl_flags_dep = dependency('absl_flags', required : false)
 absl_strings_dep = dependency('absl_strings', required : false)
 absl_synchronization_dep = dependency('absl_synchronization', required : false)
+absl_types_dep = dependency('absl_optional', required : false)
 
 # If we have the base dep, assume the rest should be present to
 if absl_base_dep.found()
@@ -55,7 +56,8 @@ if absl_base_dep.found()
     absl_base_dep,
     absl_flags_dep,
     absl_strings_dep,
-    absl_synchronization_dep
+    absl_synchronization_dep,
+    absl_types_dep,
   ]
 else
   warning('Could not find abseil-cpp with pkg-config, trying CMake-based library detection.')
@@ -63,6 +65,7 @@ else
     modules : [
       'absl::base',
       'absl::flags_parse',
+      'absl::optional',
       'absl::strings',
       'absl::synchronization',
     ],
