diff --git a/src/glibcurl.cpp b/src/glibcurl.cpp
index 98929db..7745c20 100644
--- a/src/glibcurl.cpp
+++ b/src/glibcurl.cpp
@@ -363,9 +363,11 @@ void glibcurl_init() {
   int fd;
   /* Create source object for curl file descriptors, and hook it into the
      default main context. */
-  curlSrc = (CurlGSource*)g_source_new(&curlFuncs, sizeof(CurlGSource));
-  g_source_attach(&curlSrc->source, NULL);
-
+  GSource     *gsource;
+  gsource = g_source_new(&curlFuncs, sizeof(CurlGSource));
+  g_source_set_priority(gsource, G_PRIORITY_DEFAULT_IDLE);
+  curlSrc = (CurlGSource*)gsource;
+  g_source_attach(&curlSrc->source, g_main_context_default());
   /* Init rest of our data */
   memset(&curlSrc->lastPollFd, 0, sizeof(curlSrc->lastPollFd));
   for (fd = 1; fd <= GLIBCURL_FDMAX; ++fd)
@@ -377,7 +379,6 @@ void glibcurl_init() {
   curl_global_init(CURL_GLOBAL_ALL);
   curlSrc->multiHandle = curl_multi_init();
   curl_multi_setopt(curlSrc->multiHandle, CURLMOPT_MAXCONNECTS, 4);
-  g_source_set_priority((GSource*)glibcurl_handle(), G_PRIORITY_DEFAULT_IDLE);
   D((stderr, "events: R=%x W=%x X=%x\n", GLIBCURL_READ, GLIBCURL_WRITE,
      GLIBCURL_EXC));
 
