diff -ruN a/ext/wayland/wldrmallocator.c b/ext/wayland/wldrmallocator.c
--- a/ext/wayland/wldrmallocator.c	2021-09-21 12:46:10.949683969 +0530
+++ b/ext/wayland/wldrmallocator.c	2021-09-21 12:46:38.782775348 +0530
@@ -351,7 +351,7 @@
 struct wl_buffer * gst_wl_drm_memory_construct_wl_buffer (GstWlDisplay * display, GstMemory * mem, const GstVideoInfo * vinfo)
 {
   guint width, height, size, maxsize;
-  guint stride, offset, memsize;
+  gsize stride, offset, memsize;
 
   enum wl_drm_format format;
   struct wl_buffer *buffer;
