diff -urN libmtp-1.1.6/Makefile.am libmtp-1.1.6-new/Makefile.am
--- libmtp-1.1.6/Makefile.am	2012-09-17 05:14:34.000000000 +0900
+++ libmtp-1.1.6-new/Makefile.am	2013-10-21 14:16:55.786754030 +0900
@@ -1,4 +1,4 @@
-SUBDIRS=src examples util doc
+SUBDIRS=src util doc
 ACLOCAL_AMFLAGS=-I m4
 
 pkgconfigdir=$(libdir)/pkgconfig
@@ -8,17 +8,17 @@
 
 # This stuff only makes sense on Linux so only
 # build and ship it on Linux.
-if USE_LINUX
-udevrulesdir=@UDEV@/rules.d
-udevrules_DATA=@UDEV_RULES@
-noinst_DATA=libmtp.usermap libmtp.fdi
+#if USE_LINUX
+#udevrulesdir=@UDEV@/rules.d
+#udevrules_DATA=@UDEV_RULES@
+#noinst_DATA=libmtp.usermap libmtp.fdi
 
-libmtp.usermap: util/mtp-hotplug
-	util/mtp-hotplug > libmtp.usermap
-@UDEV_RULES@: util/mtp-hotplug
-	util/mtp-hotplug -u -p"@UDEV@" @UDEV_GROUP@ @UDEV_MODE@ > @UDEV_RULES@
-libmtp.fdi: util/mtp-hotplug
-	util/mtp-hotplug -H > libmtp.fdi
+#libmtp.usermap: util/mtp-hotplug
+#	util/mtp-hotplug > libmtp.usermap
+#@UDEV_RULES@: util/mtp-hotplug
+#	util/mtp-hotplug -u -p"@UDEV@" @UDEV_GROUP@ @UDEV_MODE@ > @UDEV_RULES@
+#libmtp.fdi: util/mtp-hotplug
+#	util/mtp-hotplug -H > libmtp.fdi
 
-CLEANFILES = libmtp.usermap @UDEV_RULES@ libmtp.fdi
-endif
+#CLEANFILES = libmtp.usermap @UDEV_RULES@ libmtp.fdi
+#endif
diff -urN libmtp-1.1.6/util/Makefile.am libmtp-1.1.6-new/util/Makefile.am
--- libmtp-1.1.6/util/Makefile.am	2012-09-17 05:14:34.000000000 +0900
+++ libmtp-1.1.6-new/util/Makefile.am	2013-10-21 14:17:34.746980531 +0900
@@ -1,11 +1,11 @@
-if USE_LINUX
-bin_PROGRAMS=mtp-hotplug
-mtp_hotplug_SOURCES=mtp-hotplug.c
+#if USE_LINUX
+#bin_PROGRAMS=mtp-hotplug
+#mtp_hotplug_SOURCES=mtp-hotplug.c
 
-mtp_probedir=@UDEV@
-mtp_probe_PROGRAMS=mtp-probe
-mtp_probe_SOURCES=mtp-probe.c
-endif
+#mtp_probedir=@UDEV@
+#mtp_probe_PROGRAMS=mtp-probe
+#mtp_probe_SOURCES=mtp-probe.c
+#endif
 
 AM_CPPFLAGS=-I$(top_builddir)/src
 LDADD=../src/libmtp.la
