From 4c067484e7dc2d669538a71b19b77f5341f72985 Mon Sep 17 00:00:00 2001
From: Byungseon Shin <sun.shin@lge.com>
Date: Fri, 7 Feb 2014 04:53:25 +0100
Subject: [PATCH] WebOS platform expects filenames in UTF-8

---
 src/corelib/io/qfile.h | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/src/corelib/io/qfile.h b/src/corelib/io/qfile.h
index e6f3d942fe..b40b02fbed 100644
--- a/src/corelib/io/qfile.h
+++ b/src/corelib/io/qfile.h
@@ -49,6 +49,10 @@
 #error qfile.h must be included before any header file that defines open
 #endif
 
+#ifndef Q_OS_WEBOS
+#define Q_OS_WEBOS
+#endif
+
 QT_BEGIN_NAMESPACE
 
 class QTemporaryFile;
@@ -84,6 +88,17 @@ public:
         // note: duplicated in qglobal.cpp (qEnvironmentVariable)
         return QString::fromUtf8(localFileName).normalized(QString::NormalizationForm_C);
     }
+#else
+#if defined(Q_OS_WEBOS)
+    // WebOS platform expects filenames in UTF-8
+    static inline QByteArray encodeName(const QString &fileName)
+    {
+        return fileName.toUtf8();
+    }
+    static QString decodeName(const QByteArray &localFileName)
+    {
+        return QString::fromUtf8(localFileName);
+    }
 #else
     static inline QByteArray encodeName(const QString &fileName)
     {
@@ -93,6 +108,7 @@ public:
     {
         return QString::fromLocal8Bit(localFileName);
     }
+#endif
 #endif
     inline static QString decodeName(const char *localFileName)
         { return decodeName(QByteArray(localFileName)); }
