From 7af313300aff3a1ddbcc8643e1b75a473c881355 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Fri, 7 Feb 2014 04:56:00 +0100
Subject: [PATCH] ChangeTheGlyphCacheSize

---
 src/gui/painting/qtextureglyphcache_p.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gui/painting/qtextureglyphcache_p.h b/src/gui/painting/qtextureglyphcache_p.h
index c105e89e50..37bff7e449 100644
--- a/src/gui/painting/qtextureglyphcache_p.h
+++ b/src/gui/painting/qtextureglyphcache_p.h
@@ -60,7 +60,7 @@
 #include <private/qfontengineglyphcache_p.h>
 
 #ifndef QT_DEFAULT_TEXTURE_GLYPH_CACHE_WIDTH
-#define QT_DEFAULT_TEXTURE_GLYPH_CACHE_WIDTH 256
+#define QT_DEFAULT_TEXTURE_GLYPH_CACHE_WIDTH 2048
 #endif
 
 struct glyph_metrics_t;
