From ea9bc09436c9a499e854f0004d23ad6aac046367 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Fri, 7 Feb 2014 04:57:06 +0100
Subject: [PATCH] QOpenGL2PaintEngineEx must makeCurrent own context

---
 src/gui/opengl/qopenglpaintengine.cpp | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/gui/opengl/qopenglpaintengine.cpp b/src/gui/opengl/qopenglpaintengine.cpp
index 17dc9df619..82d3aed1cb 100644
--- a/src/gui/opengl/qopenglpaintengine.cpp
+++ b/src/gui/opengl/qopenglpaintengine.cpp
@@ -2190,6 +2190,10 @@ bool QOpenGL2PaintEngineEx::end()
     QOpenGLPaintDevicePrivate::get(d->device)->endPaint();
 
     QOpenGLContext *ctx = d->ctx;
+
+    if (ctx != QOpenGLContext::currentContext())
+        ctx->makeCurrent(ctx->surface());
+
     d->funcs.glUseProgram(0);
     d->transferMode(BrushDrawingMode);
 
@@ -2223,6 +2227,12 @@ void QOpenGL2PaintEngineEx::ensureActive()
     if (d->vao.isCreated())
         d->vao.bind();
 
+    if (ctx != QOpenGLContext::currentContext()) {
+        ctx->makeCurrent(ctx->surface());
+        ctx->d_func()->active_engine = this;
+        d->needsSync = true;
+    }
+
     if (isActive() && ctx->d_func()->active_engine != this) {
         ctx->d_func()->active_engine = this;
         d->needsSync = true;
