From 6cd37e2e0be12bf264b9f88d5773c48a7493cf3e Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Fri, 7 Feb 2014 04:59:35 +0100
Subject: [PATCH] platformsupport.pro: Add MESA_EGL_NO_X11_HEADERS

---
 src/platformsupport/platformsupport.pro | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/platformsupport/platformsupport.pro b/src/platformsupport/platformsupport.pro
index 6d4f1b93bd..0c94ba0b6e 100644
--- a/src/platformsupport/platformsupport.pro
+++ b/src/platformsupport/platformsupport.pro
@@ -11,6 +11,9 @@ SUBDIRS = \
 qtConfig(freetype)|darwin|win32: \
     SUBDIRS += fontdatabases
 
+#Avoid X11 header collision
+DEFINES += MESA_EGL_NO_X11_HEADERS
+
 qtConfig(evdev)|qtConfig(tslib)|qtConfig(libinput)|qtConfig(integrityhid) {
     SUBDIRS += input
     input.depends += devicediscovery
