From b4a2397dec64972f9b161f5f13183b0c45d12dcf Mon Sep 17 00:00:00 2001
From: JeongBong Seo <jb.seo@lge.com>
Date: Tue, 17 Dec 2013 17:33:48 +0900
Subject: [PATCH] Workaround for SGX clipping bug

---
 src/gui/opengl/qopengltextureglyphcache.cpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/gui/opengl/qopengltextureglyphcache.cpp b/src/gui/opengl/qopengltextureglyphcache.cpp
index 90897a111e..1a69a138a1 100755
--- a/src/gui/opengl/qopengltextureglyphcache.cpp
+++ b/src/gui/opengl/qopengltextureglyphcache.cpp
@@ -420,7 +420,14 @@ void QOpenGLTextureGlyphCache::resizeTextureData(int width, int height)
 
     if (pex != 0) {
         funcs->glViewport(0, 0, pex->width, pex->height);
+#ifndef QT_GL_FORCE_REGENERATE_CLIP
         pex->updateClipScissorTest();
+#else
+        if (pex->q->state()->clipTestEnabled)
+            pex->regenerateClip();
+        else
+            pex->updateClipScissorTest();
+#endif
     } else {
         if (m_vao.isCreated()) {
             m_vao.release();
