From 65c1eda3bb3d2652293a8f58ba69237ead95c686 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Fri, 7 Feb 2014 04:59:01 +0100
Subject: [PATCH] Workaround for RGX broken render to texture with size over
 8192

---
 src/gui/opengl/qopengltextureglyphcache.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/gui/opengl/qopengltextureglyphcache.cpp b/src/gui/opengl/qopengltextureglyphcache.cpp
index 728d23f61e..aba98e1425 100755
--- a/src/gui/opengl/qopengltextureglyphcache.cpp
+++ b/src/gui/opengl/qopengltextureglyphcache.cpp
@@ -470,7 +470,11 @@ int QOpenGLTextureGlyphCache::maxTextureWidth() const
     if (ctx == 0)
         return QImageTextureGlyphCache::maxTextureWidth();
     else
+#ifdef QT_GL_BROKEN_RENDER_TO_TEXTURE_OVER_8192
+        return qMin(8192, ctx->d_func()->maxTextureSize());
+#else
         return ctx->d_func()->maxTextureSize();
+#endif
 }
 
 int QOpenGLTextureGlyphCache::maxTextureHeight() const
@@ -485,7 +489,11 @@ int QOpenGLTextureGlyphCache::maxTextureHeight() const
     if (ctx->d_func()->workaround_brokenTexSubImage)
         return qMin(1024, ctx->d_func()->maxTextureSize());
     else
+#ifdef QT_GL_BROKEN_RENDER_TO_TEXTURE_OVER_8192
+        return qMin(8192, ctx->d_func()->maxTextureSize());
+#else
         return ctx->d_func()->maxTextureSize();
+#endif
 }
 
 void QOpenGLTextureGlyphCache::clear()
