From e397e7777a33a9a811c9bb2e5c7ed8d239abdd81 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Fri, 7 Feb 2014 04:59:01 +0100
Subject: [PATCH] Workaround for RGX broken render to texture with size over
 8192

---
 src/gui/opengl/qopengltextureglyphcache.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/gui/opengl/qopengltextureglyphcache.cpp b/src/gui/opengl/qopengltextureglyphcache.cpp
index 2d62bb5c1c..ab344a51bd 100755
--- a/src/gui/opengl/qopengltextureglyphcache.cpp
+++ b/src/gui/opengl/qopengltextureglyphcache.cpp
@@ -460,7 +460,11 @@ int QOpenGLTextureGlyphCache::maxTextureWidth() const
     if (ctx == 0)
         return QImageTextureGlyphCache::maxTextureWidth();
     else
+#ifdef QT_GL_BROKEN_RENDER_TO_TEXTURE_OVER_8192
+        return qMin(8192, ctx->d_func()->maxTextureSize());
+#else
         return ctx->d_func()->maxTextureSize();
+#endif
 }
 
 int QOpenGLTextureGlyphCache::maxTextureHeight() const
@@ -475,7 +479,11 @@ int QOpenGLTextureGlyphCache::maxTextureHeight() const
     if (ctx->d_func()->workaround_brokenTexSubImage)
         return qMin(1024, ctx->d_func()->maxTextureSize());
     else
+#ifdef QT_GL_BROKEN_RENDER_TO_TEXTURE_OVER_8192
+        return qMin(8192, ctx->d_func()->maxTextureSize());
+#else
         return ctx->d_func()->maxTextureSize();
+#endif
 }
 
 void QOpenGLTextureGlyphCache::clear()
