--- a/delegate/src/DelegateUtils.hpp	2021-09-01 12:09:11.816029487 +0000
+++ b/delegate/src/DelegateUtils.hpp	2021-09-01 12:08:56.823411434 +0000
@@ -294,7 +294,7 @@
             activationDesc.m_Function = armnn::ActivationFunction::ReLu;
             break;
         }
-        case kTfLiteActRelu1:
+        case kTfLiteActReluN1To1:
         {
             activationDesc.m_Function = armnn::ActivationFunction::BoundedReLu;
             activationDesc.m_A = 1.0f;
