From 53026fc9fb02ccf5d7824fbebd5246945d41ba29 Mon Sep 17 00:00:00 2001
From: Jongwoong Lee <jw.lee@lge.com>
Date: Mon, 1 Jul 2019 19:28:02 +0900
Subject: [PATCH] Add os_arch x86_64 for webos

---
Upstream-Status: Pending

 build_common/SConscript | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build_common/SConscript b/build_common/SConscript
index d5abaaf84..1868ea8d9 100644
--- a/build_common/SConscript
+++ b/build_common/SConscript
@@ -26,7 +26,7 @@ os_arch_map = {
         'armv6l', 'armv7l'
     ],
     'tizen': ['x86', 'x86_64', 'arm', 'arm-v7a', 'armeabi-v7a', 'arm64', 'armv7l'],
-    'webos': ['i586', 'i686', 'aarch64', 'arm'],
+    'webos': ['i586', 'i686', 'aarch64', 'arm', 'x86_64'],
     'android': [
         'x86', 'x86_64', 'armeabi', 'armeabi-v7a', 'armeabi-v7a-hard',
         'arm64-v8a'
-- 
2.17.1

