From a33842efa51ca3f021310e10f444afef1e779fee Mon Sep 17 00:00:00 2001
From: Tomas Mraz <tomas@openssl.org>
Date: Wed, 7 Jun 2023 13:56:54 +0200
Subject: [PATCH] Coverity 1528492: Fix possible memory leak if t == NULL

---
Upstream-Status: Backport [openssl-3.2.0-alpha1 a33842efa51ca3f021310e10f444afef1e779fee]

 crypto/asn1/a_time.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/crypto/asn1/a_time.c b/crypto/asn1/a_time.c
index 4f51a6bc4a..dd3724fc83 100644
--- a/crypto/asn1/a_time.c
+++ b/crypto/asn1/a_time.c
@@ -569,7 +569,7 @@ int ASN1_TIME_normalize(ASN1_TIME *t)
 {
     struct tm tm;
 
-    if (!ASN1_TIME_to_tm(t, &tm))
+    if (t == NULL || !ASN1_TIME_to_tm(t, &tm))
         return 0;
 
     return ossl_asn1_time_from_tm(t, &tm, V_ASN1_UNDEF) != NULL;

