From ac31ddd56ea05f2224fc7ff521c0ee49b4ea31fc Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Mon, 25 Jan 2021 14:17:35 -0800
Subject: [PATCH] add video info message

---
 .../ext/wayland/gstwaylandsink.c              | 33 +++++++++++++++++++
 1 file changed, 33 insertions(+)

diff --git a/subprojects/gst-plugins-bad/ext/wayland/gstwaylandsink.c b/subprojects/gst-plugins-bad/ext/wayland/gstwaylandsink.c
index a2a7c9b933..06c52bfe59 100644
--- a/subprojects/gst-plugins-bad/ext/wayland/gstwaylandsink.c
+++ b/subprojects/gst-plugins-bad/ext/wayland/gstwaylandsink.c
@@ -557,6 +557,34 @@ gst_wayland_create_pool (GstWaylandSink * sink, GstCaps * caps)
   return pool;
 }
 
+static gboolean
+gst_wayland_sink_post_video_info_message (GstWaylandSink * sink)
+{
+  g_return_val_if_fail (sink != NULL, FALSE);
+  g_return_val_if_fail (&sink->video_info != NULL, FALSE);
+
+  /* get video information */
+  gint width  = GST_VIDEO_INFO_WIDTH(&sink->video_info);
+  gint height = GST_VIDEO_INFO_HEIGHT(&sink->video_info);
+  gint fps_n  = GST_VIDEO_INFO_FPS_N(&sink->video_info);
+  gint fps_d  = GST_VIDEO_INFO_FPS_D(&sink->video_info);
+  gint par_n  = GST_VIDEO_INFO_PAR_N(&sink->video_info);
+  gint par_d  = GST_VIDEO_INFO_PAR_D(&sink->video_info);
+  GST_DEBUG_OBJECT (sink, "video-info width[%d] height[%d] framerate[%d/%d] pixel-aspect-ratio[%d/%d]",
+      width, height, fps_n, fps_d, par_n, par_d);
+
+  /* post an application message as video-info */
+  gboolean ret = gst_element_post_message (GST_ELEMENT_CAST (sink),
+                     gst_message_new_application (GST_OBJECT (sink),
+                         gst_structure_new ("video-info",
+                             "width", G_TYPE_INT, width,
+                             "height", G_TYPE_INT, height,
+                             "framerate", GST_TYPE_FRACTION, fps_n, fps_d,
+                             "pixel-aspect-ratio", GST_TYPE_FRACTION, par_n, par_d,
+                             NULL)));
+  return ret;
+}
+
 static gboolean
 gst_wayland_sink_set_caps (GstBaseSink * bsink, GstCaps * caps)
 {
@@ -575,6 +603,11 @@ gst_wayland_sink_set_caps (GstBaseSink * bsink, GstCaps * caps)
   format = GST_VIDEO_INFO_FORMAT (&sink->video_info);
   sink->video_info_changed = TRUE;
 
+  GST_INFO_OBJECT (sink, "Post video-info message to application!");
+  if (!gst_wayland_sink_post_video_info_message(sink)) {
+    GST_ERROR_OBJECT (sink, "Failed to post video-fino message to application!");
+  }
+
   /* create a new pool for the new caps */
   if (sink->pool)
     gst_object_unref (sink->pool);
