From 77207a965a6438287431f7b179f6f774dfcd9df2 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Fri, 4 Mar 2016 15:39:49 +0100
Subject: [PATCH] build-fixes

---
 Makefile | 21 +++++++++++----------
 1 file changed, 11 insertions(+), 10 deletions(-)

diff --git a/Makefile b/Makefile
index 7741842..19a4bd6 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 
 BINS = bin/mem-monitor bin/mem-cpu-monitor
-LIBS = lib/mallinfo.so
+LIBS = lib/libmallinfo.so.0
 
 all: $(BINS) $(LIBS)
 
@@ -13,28 +13,29 @@ distclean: clean
 tags:
 	ctags *.c *.h
 
-lib/mallinfo.so: src/mallinfo.c
+lib/libmallinfo.so.0: src/mallinfo.c
 	@mkdir -p lib
-	gcc -g -W -Wall -shared -O2 -fPIC  -Wl,-soname,mallinfo.so.0 -o $@ $^
+	$(CC) $(LDFLAGS) -g -W -Wall -shared -O2 -fPIC  -Wl,-soname,libmallinfo.so.0 -o $@ $^
+	ln -s libmallinfo.so.0 lib/libmallinfo.so
 
 bin/mem-monitor: src/mem-monitor.c src/mem-monitor-util.c
 	@mkdir -p bin
-	gcc -g -W -Wall -O2 -o $@ $+
+	$(CC) $(LDFLAGS) -g -W -Wall -O2 -o $@ $+
 
 bin/mem-cpu-monitor: src/mem-cpu-monitor.c src/sp_report.c
 	@mkdir -p bin
-	gcc -std=c99 -g -W -Wall -O2 -o $@ $+ -lspmeasure
+	$(CC) $(LDFLAGS) -std=c99 -g -W -Wall -O2 -o $@ $+ -lspmeasure
 
 install:
 	install -d  $(DESTDIR)/usr/bin
-	cp -a bin/* $(DESTDIR)/usr/bin
+	cp -R --no-dereference --preserve=mode,links -v bin/* $(DESTDIR)/usr/bin
 	install -d  $(DESTDIR)/usr/lib
-	cp -a lib/*.so  $(DESTDIR)/usr/lib
-	cp -a scripts/* $(DESTDIR)/usr/bin
+	cp -R --no-dereference --preserve=mode,links -v lib/*.so* $(DESTDIR)/usr/lib
+	cp -R --no-dereference --preserve=mode,links -v scripts/* $(DESTDIR)/usr/bin
 	install -d      $(DESTDIR)/usr/share/man/man1
-	cp -a man/*.1   $(DESTDIR)/usr/share/man/man1
+	cp -R --no-dereference --preserve=mode,links -v man/*.1   $(DESTDIR)/usr/share/man/man1
 	install -d      $(DESTDIR)/usr/share/sp-memusage-tests/
-	cp -a tests/*	$(DESTDIR)/usr/share/sp-memusage-tests/
+	cp -R --no-dereference --preserve=mode,links -v tests/*	$(DESTDIR)/usr/share/sp-memusage-tests/
 
 
 # other docs are installed by the Debian packaging
-- 
2.7.0

