From a5883920adff1f65a075b28dfb15ce1926209ae7 Mon Sep 17 00:00:00 2001
From: Rohit Anand <rohit.anand@lge.com>
Date: Mon, 30 May 2016 16:43:49 +0530
Subject: [PATCH 2/5] Sort opkg list file content to speed package removal

---
 libopkg/pkg.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/libopkg/pkg.c b/libopkg/pkg.c
index 11a9cf6..9afd824 100644
--- a/libopkg/pkg.c
+++ b/libopkg/pkg.c
@@ -1431,6 +1431,18 @@ int pkg_write_filelist(pkg_t * pkg)
     hash_table_foreach(&opkg_config->file_hash, pkg_write_filelist_helper,
                        &data);
     fclose(data.stream);
+
+    //sort opkg list file content
+    {
+        const char *argv[] = {"sort", list_file_name, "-o", list_file_name, NULL};
+        int err = 0;
+
+        err  = xsystem(argv);
+        if (err) {
+           opkg_msg(ERROR, "sort installed file list returned status %d.\n", err);
+        }
+    }
+
     free(list_file_name);
 
     pkg->state_flag &= ~SF_FILELIST_CHANGED;
-- 
2.15.1

