From 464334b6179a848b22fe2aef2f3abf860d327726 Mon Sep 17 00:00:00 2001
From: Hotaek Jung <hotaek.jung@lge.com>
Date: Mon, 12 Jul 2021 19:16:26 +0900
Subject: [PATCH] systemd-oomd: depend on swap-on

---
 units/systemd-oomd.service.in | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/units/systemd-oomd.service.in b/units/systemd-oomd.service.in
index 44f71c9e36..8dfc22356e 100644
--- a/units/systemd-oomd.service.in
+++ b/units/systemd-oomd.service.in
@@ -12,6 +12,8 @@ Description=Userspace Out-Of-Memory (OOM) Killer
 Documentation=man:systemd-oomd.service(8)
 DefaultDependencies=no
 Before=multi-user.target shutdown.target
+Wants=swap-on.service
+After=swap-on.service
 Conflicts=shutdown.target
 ConditionControlGroupController=v2
 ConditionControlGroupController=memory
