From 5127e9277f8d6b99dfb4a3d11f763c2d365b47b8 Mon Sep 17 00:00:00 2001
From: Hotaek Jung <hotaek.jung@lge.com>
Date: Tue, 23 Nov 2021 10:42:42 +0900
Subject: [PATCH] oomd: to some

Upstream-Status: Pending
---
 src/oom/oomd-util.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/oom/oomd-util.c b/src/oom/oomd-util.c
index f9f0af2d04..45c9d715a5 100644
--- a/src/oom/oomd-util.c
+++ b/src/oom/oomd-util.c
@@ -404,7 +404,7 @@ int oomd_cgroup_context_acquire(const char *path, OomdCGroupContext **ret) {
         if (r < 0)
                 return log_debug_errno(r, "Error getting cgroup memory pressure path from %s: %m", path);
 
-        r = read_resource_pressure(p, PRESSURE_TYPE_FULL, &ctx->memory_pressure);
+        r = read_resource_pressure(p, PRESSURE_TYPE_SOME, &ctx->memory_pressure);
         if (r < 0)
                 return log_debug_errno(r, "Error parsing memory pressure from %s: %m", p);
 
