From 18db0c47c704ba17668a61da475ad4323f918214 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Mon, 6 Nov 2017 17:22:45 +0000
Subject: [PATCH] Add tzset applet

Signed-off-by: Martin Jansa <martin.jansa@lge.com>

---
 coreutils/tzset.c | 57 +++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 57 insertions(+)
 create mode 100644 coreutils/tzset.c

diff --git a/coreutils/tzset.c b/coreutils/tzset.c
new file mode 100644
index 0000000..100ba8c
--- /dev/null
+++ b/coreutils/tzset.c
@@ -0,0 +1,57 @@
+/* -*- compile-command: "cd ../ && make -k"; -*- */
+/*
+ * applet to sync kernel's notion of timezone with userspace's
+ *
+ * by Eric House (eric.house@palm.com)
+ *
+ * Licensed under GPLv2 or later, see file LICENSE in this tarball for details.
+*/
+
+//config:config TZSET
+//config:      bool "tzset"
+//config:      default n
+//config:      help
+//config:        tzset set kernel timezone from userspace
+
+//applet:IF_TZSET(APPLET(tzset, BB_DIR_BIN, BB_SUID_DROP))
+
+//kbuild:lib-$(CONFIG_TZSET)     += tzset.o
+
+#include <time.h>
+#include <sys/syscall.h>
+
+#include "libbb.h"
+
+#define TZSET_OPT_VERBOSE		0x01 /* -v: */
+
+//usage:#define tzset_trivial_usage
+//usage:      "[v]"
+//usage:#define tzset_full_usage "\n"
+//usage:       "set kernel timezone from userspace"
+//usage:       "\n	-v		Verbose"
+
+
+int
+tzset_main( int argc UNUSED_PARAM, char **argv )
+{
+    bool verbose = false;
+    uint32_t opt = getopt32( argv, "v" );
+	if ( (opt & TZSET_OPT_VERBOSE) != 0 ) {
+        verbose = true;
+    }
+
+    tzset();                    /* set timezone and daylight */
+
+    if ( verbose ) {
+        fprintf( stderr, "Using vals timezone: %ld; daylight: %d\n", 
+                 timezone, daylight );
+    }
+
+	const struct timezone tz = { 
+        .tz_minuteswest = timezone/60 - 60*daylight, 
+        .tz_dsttime = 0 
+    };
+	(void)settimeofday( NULL, &tz );
+
+    return 0;
+}
