From be2e1a62fefe03ac50dc0e0594af4944b07319e0 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Mon, 6 Nov 2017 17:24:53 +0000
Subject: [PATCH] libedit: check for null before passing cmdedit_prompt to
 fputs

Signed-off-by: Martin Jansa <martin.jansa@lge.com>
---
Upstream-Status: Pending

 libbb/lineedit.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/libbb/lineedit.c b/libbb/lineedit.c
index e14c78707..0166f8f34 100644
--- a/libbb/lineedit.c
+++ b/libbb/lineedit.c
@@ -490,7 +490,9 @@ static void beep(void)
  */
 static void put_prompt_custom(bool is_full)
 {
-	fputs_stdout((is_full ? cmdedit_prompt : prompt_last_line));
+	if ( !is_full || NULL != cmdedit_prompt ) {
+		fputs_stdout((is_full ? cmdedit_prompt : prompt_last_line));
+	}
 	cursor = 0;
 	cmdedit_y = cmdedit_prmt_len / cmdedit_termw; /* new quasireal y */
 	cmdedit_x = cmdedit_prmt_len % cmdedit_termw;
