From 781fc27dfd515b4233322bad3669dde5a1409f4d Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin.jansa@lge.com>
Date: Mon, 6 Nov 2017 17:24:53 +0000
Subject: [PATCH] libedit: check for null before passing cmdedit_prompt to
 fputs

Signed-off-by: Martin Jansa <martin.jansa@lge.com>

---
 libbb/lineedit.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/libbb/lineedit.c b/libbb/lineedit.c
index fbabc6c..c47bddc 100644
--- a/libbb/lineedit.c
+++ b/libbb/lineedit.c
@@ -452,7 +452,9 @@ static void beep(void)
  */
 static void put_prompt_custom(bool is_full)
 {
-	fputs((is_full ? cmdedit_prompt : prompt_last_line), stdout);
+	if ( !is_full || NULL != cmdedit_prompt ) {
+		fputs((is_full ? cmdedit_prompt : prompt_last_line), stdout);
+	}
 	cursor = 0;
 	cmdedit_y = cmdedit_prmt_len / cmdedit_termw; /* new quasireal y */
 	cmdedit_x = cmdedit_prmt_len % cmdedit_termw;
