Index: busybox-1.19.4/coreutils/date.c
===================================================================
--- busybox-1.19.4.orig/coreutils/date.c	2012-07-23 16:49:41.000000000 -0700
+++ busybox-1.19.4/coreutils/date.c	2012-07-23 16:58:21.000000000 -0700
@@ -103,6 +103,7 @@
 //usage:     "\n	-S		Get/set time as seconds since Epoc"
 //usage:     "\n	-u		Work in UTC (don't convert to local time)"
 //usage:     "\n	-R		Output RFC-2822 compliant date string"
+//usage:     "\n	-U		Only output uptime, seconds since system start"
 //usage:	) IF_LONG_OPTS(
 //usage:     "\n	[-s,--set] TIME	Set time to TIME"
 //usage:     "\n	-S,--seconds	Get/set time as seconds since Epoc"
@@ -137,9 +138,7 @@
 //usage:       "Wed Apr 12 18:52:41 MDT 2000\n"
 
 #include "libbb.h"
-#if ENABLE_FEATURE_DATE_NANO
-# include <sys/syscall.h>
-#endif
+#include <sys/syscall.h>
 
 enum {
 	OPT_RFC2822   = (1 << 0), /* R */
@@ -148,8 +147,9 @@
 	OPT_DATE      = (1 << 3), /* d */
 	OPT_REFERENCE = (1 << 4), /* r */
 	OPT_SECONDS   = (1 << 5), /* S */
-	OPT_TIMESPEC  = (1 << 6) * ENABLE_FEATURE_DATE_ISOFMT, /* I */
-	OPT_HINT      = (1 << 7) * ENABLE_FEATURE_DATE_ISOFMT, /* D */
+        OPT_UPTIME    = (1 << 6), /* U */
+	OPT_TIMESPEC  = (1 << 7) * ENABLE_FEATURE_DATE_ISOFMT, /* I */
+	OPT_HINT      = (1 << 8) * ENABLE_FEATURE_DATE_ISOFMT, /* D */
 };
 
 static void maybe_set_utc(int opt)
@@ -187,7 +187,7 @@
 	opt_complementary = "d--s:s--d"
 		IF_FEATURE_DATE_ISOFMT(":R--I:I--R");
 	IF_LONG_OPTS(applet_long_options = date_longopts;)
-	opt = getopt32(argv, "Rs:ud:r:S"
+	opt = getopt32(argv, "Rs:ud:r:SU"
 			IF_FEATURE_DATE_ISOFMT("I::D:"),
 			&date_str, &date_str, &filename
 			IF_FEATURE_DATE_ISOFMT(, &isofmt_arg, &fmt_str2dt));
@@ -258,6 +258,14 @@
 		 * drop a mail to project mailing list please
 		 */
 #endif
+        } else if ( opt & OPT_UPTIME) {
+                struct timespec tp;
+		if ( syscall(__NR_clock_gettime, CLOCK_MONOTONIC, &ts) ) {
+                       bb_perror_msg("cannot get date");
+                       return EXIT_FAILURE;
+                }
+                printf("%d.%02ld\n", ts.tv_sec, ts.tv_nsec / 10000000);
+                return EXIT_SUCCESS;
 	} else {
 #if ENABLE_FEATURE_DATE_NANO
 		/* libc has incredibly messy way of doing this,
