Index: b/coreutils/Config.src
===================================================================
--- a/coreutils/Config.src	2012-07-23 15:40:07.000000000 -0700
+++ b/coreutils/Config.src	2012-07-23 15:40:34.000000000 -0700
@@ -668,6 +668,12 @@
 	  tty is used to print the name of the current terminal to
 	  standard output.
 
+config TZSET
+	bool "tzset"
+	default n
+	help
+	  tzset set kernel timezone from userspace
+
 config UNAME
 	bool "uname"
 	default y
Index: b/coreutils/Kbuild.src
===================================================================
--- a/coreutils/Kbuild.src	2012-07-23 15:38:53.000000000 -0700
+++ b/coreutils/Kbuild.src	2012-07-23 15:39:09.000000000 -0700
@@ -75,6 +75,7 @@
 lib-$(CONFIG_TEE)       += tee.o
 lib-$(CONFIG_TRUE)      += true.o
 lib-$(CONFIG_TTY)       += tty.o
+lib-$(CONFIG_TZSET)     += tzset.o
 lib-$(CONFIG_UNAME)     += uname.o
 lib-$(CONFIG_UNEXPAND)  += expand.o
 lib-$(CONFIG_UNIQ)      += uniq.o
Index: b/coreutils/tzset.c
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ b/coreutils/tzset.c	2012-07-23 15:46:32.000000000 -0700
@@ -0,0 +1,47 @@
+/* -*- compile-command: "cd ../ && make -k"; -*- */
+/*
+ * applet to sync kernel's notion of timezone with userspace's
+ *
+ * by Eric House (eric.house@palm.com)
+ *
+ * Licensed under GPLv2 or later, see file LICENSE in this tarball for details.
+*/
+
+#include <time.h>
+#include <sys/syscall.h>
+
+#include "libbb.h"
+
+#define TZSET_OPT_VERBOSE		0x01 /* -v: */
+
+//usage:#define tzset_trivial_usage
+//usage:      "[v]"
+//usage:#define tzset_full_usage "\n"
+//usage:       "set kernel timezone from userspace"
+//usage:       "\n	-v		Verbose"
+
+
+int
+tzset_main( int argc UNUSED_PARAM, char **argv )
+{
+    bool verbose = false;
+    uint32_t opt = getopt32( argv, "v" );
+	if ( (opt & TZSET_OPT_VERBOSE) != 0 ) {
+        verbose = true;
+    }
+
+    tzset();                    /* set timezone and daylight */
+
+    if ( verbose ) {
+        fprintf( stderr, "Using vals timezone: %ld; daylight: %d\n", 
+                 timezone, daylight );
+    }
+
+	const struct timezone tz = { 
+        .tz_minuteswest = timezone/60 - 60*daylight, 
+        .tz_dsttime = 0 
+    };
+	(void)settimeofday( NULL, &tz );
+
+    return 0;
+}
Index: b/include/applets.src.h
===================================================================
--- a/include/applets.src.h	2012-07-23 15:35:17.000000000 -0700
+++ b/include/applets.src.h	2012-07-23 15:35:28.000000000 -0700
@@ -381,6 +381,7 @@
 IF_TTYSIZE(APPLET(ttysize, BB_DIR_USR_BIN, BB_SUID_DROP))
 IF_TUNCTL(APPLET(tunctl, BB_DIR_SBIN, BB_SUID_DROP))
 IF_TUNE2FS(APPLET(tune2fs, BB_DIR_SBIN, BB_SUID_DROP))
+IF_TZSET(APPLET(tzset, BB_DIR_BIN, BB_SUID_DROP))
 IF_UDHCPC(APPLET(udhcpc, BB_DIR_SBIN, BB_SUID_DROP))
 IF_UDHCPD(APPLET(udhcpd, BB_DIR_USR_SBIN, BB_SUID_DROP))
 IF_UDPSVD(APPLET_ODDNAME(udpsvd, tcpudpsvd, BB_DIR_USR_BIN, BB_SUID_DROP, udpsvd))
