Index: busybox-1.19.4/libbb/lineedit.c
===================================================================
--- busybox-1.19.4.orig/libbb/lineedit.c	2012-07-23 17:07:09.000000000 -0700
+++ busybox-1.19.4/libbb/lineedit.c	2012-07-23 17:07:58.000000000 -0700
@@ -429,7 +429,9 @@
 {
 	unsigned w;
 
-	fputs(cmdedit_prompt, stdout);
+	if ( NULL != cmdedit_prompt ) {
+	      fputs(cmdedit_prompt, stdout);
+	}
 	fflush_all();
 	cursor = 0;
 	w = cmdedit_termw; /* read volatile var once */
