diff --git a/CMakeLists.txt b/CMakeLists.txt
index c61636a2..3e89db3b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -54,7 +54,7 @@ option(FLB_PARSER             "Build with Parser support"    Yes)
 option(FLB_TLS                "Build with SSL/TLS support"    No)
 option(FLB_BINARY             "Build executable binary"      Yes)
 option(FLB_EXAMPLES           "Build examples"               Yes)
-option(FLB_SHARED_LIB         "Build shared library"         Yes)
+set(FLB_SHARED_LIB         "Build shared library"         Yes)
 option(FLB_VALGRIND           "Enable Valgrind support"       No)
 option(FLB_TRACE              "Enable trace mode"             No)
 option(FLB_TESTS_RUNTIME      "Enable runtime tests"          No)
@@ -302,7 +302,7 @@ option(MSGPACK_ENABLE_CXX             OFF)
 option(MSGPACK_ENABLE_SHARED          OFF)
 option(MSGPACK_BUILD_TESTS            OFF)
 option(MSGPACK_BUILD_EXAMPLES         OFF)
-add_subdirectory(${FLB_PATH_LIB_MSGPACK} EXCLUDE_FROM_ALL)
+add_subdirectory(${FLB_PATH_LIB_MSGPACK} )
 
 # MPack
 add_subdirectory(${FLB_PATH_LIB_MPACK} EXCLUDE_FROM_ALL)
@@ -336,7 +336,7 @@ MK_SET_OPTION(MK_DEBUG          ON)
 if(FLB_HTTP_SERVER)
   add_subdirectory(${FLB_PATH_LIB_MONKEY} EXCLUDE_FROM_ALL)
 else()
-  add_subdirectory(${FLB_PATH_LIB_MONKEY}/mk_core EXCLUDE_FROM_ALL)
+  add_subdirectory(${FLB_PATH_LIB_MONKEY}/mk_core)
 endif()
 
 if(FLB_TLS)
@@ -344,7 +344,7 @@ if(FLB_TLS)
   option(ENABLE_TESTING  OFF)
   option(ENABLE_PROGRAMS OFF)
   option(INSTALL_MBEDTLS_HEADERS OFF)
-  add_subdirectory(${FLB_PATH_LIB_MBEDTLS} EXCLUDE_FROM_ALL)
+  add_subdirectory(${FLB_PATH_LIB_MBEDTLS})
 endif()
 
 # Metrics
@@ -633,6 +633,7 @@ else()
 endif()
 
 # Instruct CMake to build the Fluent Bit Core
+add_subdirectory(lib/monkey/include)
 add_subdirectory(include)
 add_subdirectory(plugins)
 add_subdirectory(src)
diff --git a/lib/flb_libco/CMakeLists.txt b/lib/flb_libco/CMakeLists.txt
index 786350de..4e1064a4 100644
--- a/lib/flb_libco/CMakeLists.txt
+++ b/lib/flb_libco/CMakeLists.txt
@@ -4,3 +4,16 @@ set(src
 
 add_definitions(-DLIBCO_MP)
 add_library(co STATIC ${src})
+
+set(INSTALL_LIBCO_HEADERS "Install libco headers." ON)
+
+
+if(INSTALL_LIBCO_HEADERS)
+    file(GLOB headers "*.h")
+
+    install(FILES ${headers}
+        DESTINATION include/
+        PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ)
+
+endif(INSTALL_LIBCO_HEADERS)
+
diff --git a/lib/jsmn/CMakeLists.txt b/lib/jsmn/CMakeLists.txt
index 3ee5421f..8b1060e9 100644
--- a/lib/jsmn/CMakeLists.txt
+++ b/lib/jsmn/CMakeLists.txt
@@ -4,3 +4,13 @@ set(src
 
 SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -fPIC -DJSMN_PARENT_LINKS -DJSMN_STRICT")
 add_library(jsmn STATIC ${src})
+
+set(INSTALL_JSMN_HEADERS "Install jsmn headers." ON)
+
+if(INSTALL_JSMN_HEADERS)
+    file(GLOB headers "*.h")
+
+    install(FILES ${headers}
+        DESTINATION include/jsmn
+        PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ)
+endif(INSTALL_JSMN_HEADERS)
diff --git a/lib/mbedtls-2.16.3/include/CMakeLists.txt b/lib/mbedtls-2.16.3/include/CMakeLists.txt
index c2f2bd4e..fc0efaaa 100644
--- a/lib/mbedtls-2.16.3/include/CMakeLists.txt
+++ b/lib/mbedtls-2.16.3/include/CMakeLists.txt
@@ -1,4 +1,4 @@
-option(INSTALL_MBEDTLS_HEADERS "Install mbed TLS headers." ON)
+set(INSTALL_MBEDTLS_HEADERS "Install mbed TLS headers." ON)
 
 if(INSTALL_MBEDTLS_HEADERS)
 
diff --git a/lib/monkey/include/CMakeLists.txt b/lib/monkey/include/CMakeLists.txt
index 4175d365..aca5b680 100644
--- a/lib/monkey/include/CMakeLists.txt
+++ b/lib/monkey/include/CMakeLists.txt
@@ -1,18 +1,11 @@
-if(NOT MK_LOCAL)
+set(MK_PATH_HEADERS ${CMAKE_INSTALL_INCLUDEDIR}/monkey CACHE STRING "Server header files (development)")
+
   install(DIRECTORY
       ${CMAKE_CURRENT_SOURCE_DIR}/monkey/
     DESTINATION
       ${MK_PATH_HEADERS})
-endif()
-
-# MK_CORE
-if(NOT WITHOUT_HEADERS)
-  install(FILES "mk_core.h"
-    DESTINATION include/
-    PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ)
 
   file(GLOB headers "mk_core/*.h")
   install(FILES ${headers}
     DESTINATION include/mk_core
     PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ)
-endif()
diff --git a/lib/monkey/include/monkey/mk_core/mk_list.h b/lib/monkey/include/monkey/mk_core/mk_list.h
index a83be12e..028180bb 100644
--- a/lib/monkey/include/monkey/mk_core/mk_list.h
+++ b/lib/monkey/include/monkey/mk_core/mk_list.h
@@ -35,7 +35,7 @@
 #endif
 
 #define container_of(ptr, type, member) ({                      \
-      const typeof( ((type *)0)->member ) *__mptr = (ptr);      \
+      const __typeof__( ((type *)0)->member ) *__mptr = (ptr);      \
       (type *)( (char *)__mptr - offsetof(type,member) );})
 #endif
 
