diff -ruN a/src/gallium/state_trackers/dri/dri2.c b/src/gallium/state_trackers/dri/dri2.c
--- a/src/gallium/state_trackers/dri/dri2.c	2019-10-26 16:27:07.153628447 +0530
+++ b/src/gallium/state_trackers/dri/dri2.c	2019-10-26 16:27:54.294874608 +0530
@@ -1287,8 +1287,10 @@
               int *fds, int num_fds, int *strides, int *offsets,
               void *loaderPrivate)
 {
+   uint64_t modifier = (fourcc == __DRI_IMAGE_FOURCC_RGB565) ?
+       DRM_FORMAT_MOD_LINEAR : DRM_FORMAT_MOD_INVALID;
    return dri2_create_image_from_fd(screen, width, height, fourcc,
-                                   DRM_FORMAT_MOD_INVALID, fds, num_fds,
+                                   modifier, fds, num_fds,
                                    strides, offsets, NULL, loaderPrivate);
 }
 
