diff -ruN a/src/gallium/state_trackers/dri/dri2.c b/src/gallium/state_trackers/dri/dri2.c
--- a/src/gallium/state_trackers/dri/dri2.c	2020-08-13 05:19:20.997163019 +0000
+++ b/src/gallium/state_trackers/dri/dri2.c	2020-08-13 05:26:16.772877920 +0000
@@ -1378,7 +1378,10 @@
    unsigned int external_only= 0;
    int count= 0;
    boolean result;
-   result= dri2_query_dma_buf_modifiers( screen, fourcc, 1, &modifier, &external_only, &count);
+	 /*For RGB16 format data since we are getting correct value of modifer as DRM_FORMAT_MOD_LINEAR
+		* no need for query of modifiers*/
+   if(modifier != DRM_FORMAT_MOD_LINEAR)
+	 result= dri2_query_dma_buf_modifiers( screen, fourcc, 1, &modifier, &external_only, &count);
    return dri2_create_image_from_fd(screen, width, height, fourcc,
                                    modifier, fds, num_fds,
                                    strides, offsets, NULL, loaderPrivate);
