--- a/delegate/CMakeLists.txt	2021-09-03 08:03:00.531634017 +0000
+++ b/delegate/CMakeLists.txt	2021-09-03 08:04:15.766594331 +0000
@@ -66,9 +66,11 @@
 target_link_libraries(armnnDelegate PUBLIC Armnn::Armnn)
 
 ## Add TfLite v2.3.1 dependency
-find_package(TfLite REQUIRED MODULE)
+#find_package(TfLite REQUIRED MODULE)
+find_package(PkgConfig)
+pkg_check_modules(TfLite REQUIRED tensorflowlite)
 
-target_link_libraries(armnnDelegate PUBLIC ${TfLite_LIB})
+target_link_libraries(armnnDelegate PUBLIC ${TfLite_LIBRARIES})
 
 # Various tflite header files are not warning clean
 # We can't change compilation flags on header files directly, so we need to add them to an interface library first
